applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -567,6 +598,8 @@ impl InnerService {
             "Difficulties: Tari ({}), Monero({}), Selected({})", tari_difficulty, monero_difficulty, mining_difficulty
         );
         monerod_resp["result"]["difficulty"] = mining_difficulty.into();
+        monerod_resp["result"]["difficulty_tari"] = tari_difficulty.into();
+        monerod_resp["result"]["difficulty_monero"] = monero_difficulty.into();
```

Comment:
Reason for adding these to the response? They will not be used anywhere by monero related software.

Should we change the rpc interface, we will have to do it in a way that is generic and possibly submit changes to the monero codebase to account for them.

Reply:
There is currently no way to tell which difficulty was achieved.

JSON is a pretty flexible format, adding fields cannot break any API consumer so these are non breaking changes.

My thoughts are that we can have a consistent key for MM proxy extensions.

Something like

```javascript
{
  "id": 123,
  ... etc
  "result": {
    "difficulty": ....,
    "_mergeMining": { // underscore because non-standard
      "difficulty_tari": ....,
      "difficulty_monero": ..... 
   }
  }
}

Reply:
Through the logs we can tell which difficulty was targeted.

Is it definitely needed since we would be submitting to the proxy at least if the template difficulty was achieved?

Template difficulty through the proxy will always be `min(tari,monero)` in the current code.

I do agree with your suggestion of using underscores to denote non-standard fields, well thought out.

Reply:
In the pool context, we need to keep track of the payout accounting for monero and tari. So we need some way to tell which block was won. We could also add a custom extension call that gives us this info, but I feel that since get_block_template gives you the required mining difficulty, it may as well provide metadata for difficulties on the merge mined and parent chains. Open to thoughts on this though.